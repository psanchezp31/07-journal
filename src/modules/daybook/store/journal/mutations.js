// export const myMutation = (state) =>{
// }

export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries];
  state.isLoading = false;
};
export const updateEntry = (state, updatedEntry) => {
  const idx = state.entries.map((e) => e.id).indexOf(updatedEntry.id);
  state.entries[idx] = updatedEntry
};

export const addEntry = (state, createdEntry) => {
  // state.entries.push(createdEntry)
  state.entries = [createdEntry, ...state.entries] // en el nuevo arreglo, ahora irÃ¡ 
  //la nueva entrada de primero y el resto van despues
};

export const deleteEntry = (state, id) =>{
  state.entries = state.entries.filter(entry => entry.id !== id)
}

export const clearEntries = (state) =>{
  state.entries = []
}
