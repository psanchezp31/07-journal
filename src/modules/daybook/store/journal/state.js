export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toDateString(),
      text: "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptates,quasi at. Atque, deleniti digniss",
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toDateString(),
      text: "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptates,quasi at. Atque, deleniti digniss",
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toDateString(),
      text: "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptates,quasi at. Atque, deleniti digniss",
      picture: null,
    },
  ],
});
