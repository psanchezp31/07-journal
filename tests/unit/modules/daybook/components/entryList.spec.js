import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";
import journal from "@/modules/daybook/store/journal";

import EntryList from "@/modules/daybook/components/EntryList.vue";
import { journalState } from "../../../mock-data/test-journal-state";

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });
describe("pruebas en el EntryList component", () => {
  const store = createVuexStore(journalState);
  const mockRouter = {
    push: jest.fn(),
  };
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks(); //para que limpie las pruebas cada vez que termine cada prueba
    wrapper = shallowMount(EntryList, {
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  test("debe de llamar el getEntriesByTerm sin término y mostrar dos entradas ", () => {
    expect(wrapper.findAll("entry-stub").length).toBe(2);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe de llamar el getEntriesByTerm sin término", async () => {
    const input = wrapper.find("input");
    await input.setValue("Mannyto");

    expect(wrapper.findAll("entry-stub").length).toBe(1);
  });

  test("el boton de nuevo debe de redireccionar a /new", () => {
    wrapper.find("button").trigger("click");
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: "new" },
    });
  });
});
