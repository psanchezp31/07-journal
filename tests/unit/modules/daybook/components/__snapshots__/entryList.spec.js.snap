// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el EntryList component debe de llamar el getEntriesByTerm sin t√©rmino y mostrar dos entradas  1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar entrada"></div>
  <div class="mt-2 d-flex"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
